code/_globalvars/lists/keybinding.dm

```diff
@@ -4,15 +4,18 @@
 		var/datum/keybinding/keybinding = KB
 		if(!initial(keybinding.keybind_signal))
 			continue
+
 		add_keybinding(new keybinding)
 	init_emote_keybinds()
 
 
+
 /// Adds an instanced keybinding to the global tracker
 /proc/add_keybinding(datum/keybinding/instance)
 	if(!instance.name)
 		return
 	GLOB.keybindings_by_name[instance.name] = instance
+	GLOB.keybindings_by_signal[instance.keybind_signal] = instance
```

Comment:
Just add a quick glance at the pr and not at current kebind code, but I find it weird that you need two glob lists and not just one. 

Cant you use current glob list ? Or refactor slightly where that lost is used to only have one list ?

Reply:
I need the keybindings_by_signal because the action datums hold only the signal to trigger , but not the name.

Reply:
Yes it doesnt hold, but maybe it should then

Reply:
maybe ? but  it doesn't add anything to the process really , the keybinding datums don't hold said keybind for any client , so it still needs to iterate through the controlling client's preference keybinds.